.github/workflows/cleanup-azure-resources.yml

```diff
@@ -1,37 +1,120 @@
+name: CleanupAzureResources
+
 on:
   pull_request:
     types: [closed]
 
-name: CleanupAzureResources
+permissions:
+   id-token: write
+   contents: write
 
 jobs:
   cleanup:
-    runs-on: ubuntu-latest
+    runs-on: windows-latest # AzureAD PS module requires Windows.
+
     steps:
-      - name: Login via Az module
+      - name: Checkout Current Branch to import custom Powershell Modules
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: 'main'
+
+      - name: Azure Login
         uses: azure/login@v1.4.1
         with:
-          creds: ${{ secrets.AZURE_CREDENTIALS }}
+          client-id: '1e8ef5b4-c0cb-428a-a9b4-fc532567bb9d' #${{ secrets.AZ_DEVOPS_APP_ID }}
+          tenant-id: '72f988bf-86f1-41af-91ab-2d7cd011db47' #${{ secrets.AZ_DEVOPS_TENANT_ID }}
+          subscription-id: 'a1766500-6fd5-4f5c-8515-607798271014' #${{ secrets.AZ_DEVOPS_SUBSCRIPTION_ID }}
           enable-AzPSSession: true
-      
-      - name: Clean Up Azure Resources
+
+      - name: Clean Up Azure Resource Groups
         uses: azure/powershell@v1.1.0
         with:
-          azPSVersion: '3.1.0'
+          azPSVersion: latest
           inlineScript: | 
-            [string]$PRNumber = "${{github.event.pull_request.number}}"
-            $groups = Get-AzResourceGroup -Name "dcm-pr$PRNumber-*-*"  
+            $azContext = Get-AzContext
+
+            [string]$prNumber = "${{ github.event.pull_request.number }}"
+            [string]$resourceGroupName = "dcm-pr$prNumber-*-*"
+
+            $groups = Get-AzResourceGroup -DefaultProfile $azContext -Name $resourceGroupName
             if ($groups -eq $null -or $groups.Count -eq 0)
             {
-              Write-Host "No Resource groups related to PR $PRNumber are found."
+              Write-Host "No Resource groups related to PR $prNumber are found."
             }
             else
             {
-              Write-Host "$($groups.Count) Resource groups related to PR $PRNumber are found."
+              Write-Host "$($groups.Count) Resource groups related to PR $prNumber are found."
+            
               foreach($group in $groups)
               {
+                Write-Host "$($groups.Count) Resource groups related to PR $prNumber are found."
                 [string]$name = $group.ResourceGroupName
                 Write-Host "Deleting Resource Group $name"
-                Remove-AzResourceGroup -Name $name -Force
-              }  
+                Remove-AzResourceGroup -DefaultProfile $azContext -Name $name -Force
+              }
+            }
+
+      - name: Purge key vault
+        uses: azure/powershell@v1.1.0
+        with:
+          azPSVersion: latest
+          inlineScript: | 
+            $azContext = Get-AzContext
+
+            [string]$resourceGroupRegion = 'southcentralus'
```

Comment:
Is the region hardcoded in the PR pipeline?

Reply:
Yes. In DevOps we have these in a variables yaml.